### ISSUE TITLE
Files in the publisher library are missing a file description field

### ISSUE THREAD

**Issue by mtub** [pkp/pkp-lib#10680]
## Issue Description
Editors cannot add file descriptions to files uploaded to the publisher library. Having file descriptions would improve accessibility, because it would allow users to understand what certain files are about. For images, these file descriptions could be used for/as alt-texts.

Since the publisher library can be used for workflow diagrams, logos, and many more images, adding this feature would improve alt-text availabilty a lot.

## How to reproduce / Current behavior
Currently, the publisher library only asks for `name` and `type` of uploaded files.

![Image](https://github.com/user-attachments/assets/2f90070c-3bc3-459b-aeca-7c3a9bb52a18)

## What application are you using?
OJS 3.3

## Action
Add a field for file descriptions. This should be mandatory, at least for images.

**Comment by NateWr** [comment]
@asmecher this would be useful for some client work I am doing with TIB (@mtub) and SLUB Dresden (@basti95), which will result in a community plugin to display partner logos (see #10521). 

Would you be open to a PR on this for 3.3? 

(I don't _think_ this would require a database migration, since the description would be localized and there is already a `library_file_settings` table.)

**Comment by asmecher** [comment]
If there's no database migration required, and you don't mind coding forward to other branches, I'd be open to a PR that goes all the way back to 3.3.

**Comment by NateWr** [comment]
@asmecher I added the description field to Publisher and Submission Library files.

PR:
https://github.com/pkp/pkp-lib/pull/10708 (`stable-3_3_0`)
https://github.com/pkp/pkp-lib/pull/10709 (`stable-3_4_0`)
https://github.com/pkp/pkp-lib/pull/10710 (`main`)

TESTS:
https://github.com/pkp/ojs/pull/4555 (`stable-3_3_0`)
https://github.com/pkp/ojs/pull/4556 (`stable-3_4_0`)
https://github.com/pkp/ojs/pull/4557 (`main`)

**Comment by NateWr** [comment]
@asmecher just following up on these PRs. I've had a look at the test failures and they don't look related to the code change. They are mostly related to postgres and the error seems like maybe a config problem:

```
Uncaught PDOException: SQLSTATE[HY000] [2002] No such file or directory in /home/runner/ojs/lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Database/Connectors/Connector.php:70
Stack trace:
#0 /home/runner/ojs/lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Database/Connectors/Connector.php(70): PDO->__construct()
#1 /home/runner/ojs/lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Database/Connectors/Connector.php(100): Illuminate\Database\Connectors\Connector->createPdoConnection()
#2 /home/runner/ojs/lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Database/Connectors/Connector.php(50): Illuminate\Database\Connectors\Connector->tryAgainIfCausedByLostConnection()
#3 /home/runner/ojs/lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Database/Connectors/MySqlConnector.php(24): Illuminate\Database\Connectors\Connector->createConnection()
#4 /home/runner/ojs/lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Database/Connectors/ConnectionFactory.php(184): Illuminate\Database\Connectors\MySqlConnector->connect()
#5 [internal function]: Illuminate\Database\Connectors\ConnectionFactory->Illuminate\Database\Connectors\{closure}()
#6 /home/runner/ojs/lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Database/Connection.php(1181): call_user_func()
#7 /home/runner/ojs/lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Database/Connection.php(1217): Illuminate\Database\Connection->getPdo()
#8 /home/runner/ojs/lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Database/Connection.php(486): Illuminate\Database\Connection->getReadPdo()
#9 /home/runner/ojs/lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Database/Connection.php(443): Illuminate\Database\Connection->getPdoForSelect()
#10 /home/runner/ojs/lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Database/Connection.php(753): Illuminate\Database\Connection->Illuminate\Database\{closure}()
#11 /home/runner/ojs/lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Database/Connection.php(899): Illuminate\Database\Connection->runQueryCallback()
#12 /home/runner/ojs/lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Database/Connection.php(879): Illuminate\Database\Connection->tryAgainIfCausedByLostConnection()
#13 /home/runner/ojs/lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Database/Connection.php(723): Illuminate\Database\Connection->handleQueryException()
#14 /home/runner/ojs/lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Database/Connection.php(456): Illuminate\Database\Connection->run()
#15 [internal function]: Illuminate\Database\Connection->cursor()
#16 /home/runner/ojs/lib/pkp/classes/context/ContextDAO.php(99): Generator->current()
#17 /home/runner/ojs/lib/pkp/classes/core/PKPRouter.php(194): PKP\context\ContextDAO->getByPath()
#18 [internal function]: PKP\core\PKPRouter->getContext()
#19 /home/runner/ojs/lib/pkp/classes/core/PKPRequest.php(877): call_user_func_array()
#20 /home/runner/ojs/lib/pkp/classes/core/PKPRequest.php(754): PKP\core\PKPRequest->_delegateToRouter()
#21 /home/runner/ojs/classes/core/Request.php(41): PKP\core\PKPRequest->getContext()
#22 /home/runner/ojs/lib/pkp/classes/plugins/LazyLoadPlugin.php(120): APP\core\Request->getContext()
#23 /home/runner/ojs/lib/pkp/classes/plugins/LazyLoadPlugin.php(76): PKP\plugins\LazyLoadPlugin->getCurrentContextId()
#24 /home/runner/ojs/plugins/generic/dublinCoreMeta/DublinCoreMetaPlugin.php(32): PKP\plugins\LazyLoadPlugin->getEnabled()
#25 /home/runner/ojs/lib/pkp/classes/plugins/PluginRegistry.php(76): APP\plugins\generic\dublinCoreMeta\DublinCoreMetaPlugin->register()
#26 /home/runner/ojs/lib/pkp/classes/plugins/PluginRegistry.php(121): PKP\plugins\PluginRegistry::register()
#27 [internal function]: PKP\plugins\PluginRegistry::PKP\plugins\{closure}()
#28 /home/runner/ojs/lib/pkp/classes/plugins/PluginRegistry.php(121): array_walk_recursive()
#29 /home/runner/ojs/lib/pkp/classes/core/Dispatcher.php(155): PKP\plugins\PluginRegistry::loadCategory()
#30 /home/runner/ojs/lib/pkp/classes/core/PKPApplication.php(395): PKP\core\Dispatcher->dispatch()
#31 /home/runner/ojs/index.php(21): PKP\core\PKPApplication->execute()
#32 {main}
```

**Comment by asmecher** [comment]
All merged, thanks, @NateWr -- the test failure is unrelated.
